packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -799,6 +828,10 @@ async function removeOldProjectFiles(projectPath: string): Promise<void> {
   await fs.remove(path.join(fx, "new.env.default.json"));
   // version <= 2.4.1, remove .fx/appPackage.
   await fs.remove(path.join(fx, AppPackageFolderName));
+  // version <= 3.2.1
+  await fs.remove(path.join(fx, REMOTE_MANIFEST));
```

Comment:
Should we first determine the toolkit version that created the manifest and then delete files only of specific versions?
For example, if we know the project are from 2.10.0, do not delete `.fx/manifest.source.json` or `.fx/color.png`, etc.

This is to prevent accidental deletion of `.fx/color.png`.

For example, if `.fx/color.png`, `appPackage/manifest.source.json` and `appPackage/color.png` exist,  `appPackage/color.png` is correctly backed up, but  `.fx/color.png` will be deleted without backup.

Reply:
offline sync

Reply:
Offline discussed. It is hard to determine the version. And it is  very rare to happen.